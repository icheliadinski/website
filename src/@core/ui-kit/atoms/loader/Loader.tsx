import React from 'react'
import classNames from 'classnames'
import styles from './Loader.module.scss'

interface Props {
  children?: string
  isCentered?: boolean
}

export const Loader: React.FC<Props> = ({
  children = 'Loading',
  isCentered = true,
}) => {
  const classes = classNames(styles.loader, {
    [styles.centered]: isCentered,
  })
  return <div className={classes}>{children}</div>
}
