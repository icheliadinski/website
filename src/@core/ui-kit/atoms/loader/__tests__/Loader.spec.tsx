import React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import { Loader } from '../Loader'

describe('[Loader]', () => {
  let wrapper: ShallowWrapper

  beforeEach(() => {
    wrapper = shallow(<Loader />)
  })

  it('renders successfully', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('when label set', () => {
    beforeEach(() => {
      wrapper.setProps({ label: 'New Label' })
    })

    it('renders successfully', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when is not centered', () => {
    beforeEach(() => {
      wrapper.setProps({ isCentered: false })
    })

    it('renders successfully', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
