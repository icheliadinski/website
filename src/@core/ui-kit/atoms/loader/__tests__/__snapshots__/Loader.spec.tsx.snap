// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Loader] renders successfully 1`] = `
<div
  className="loader centered"
>
  Loading
</div>
`;

exports[`[Loader] when is not centered renders successfully 1`] = `
<div
  className="loader"
>
  Loading
</div>
`;

exports[`[Loader] when label set renders successfully 1`] = `
<div
  className="loader centered"
>
  Loading
</div>
`;
